jerry-ext/handler/handler-assert.c

```diff
@@ -132,3 +186,234 @@ jerryx_handler_assert (const jerry_value_t func_obj_val, /**< function object */
+ *         otherwise - this function calls jerryx_fatal.
+ *         Note that in this case the function does not return.
+ */
+static jerry_value_t
+jerryx_handler_assert_not_equal (const jerry_value_t func_obj_val, /**< function object */
+                                 const jerry_value_t this_p, /**< this arg */
+                                 const jerry_value_t args_p[], /**< function arguments */
+                                 const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  return jerryx_handler_assert_equal_helper (func_obj_val, this_p, args_p, args_cnt, JERRYX_OP_NOT_EQUAL);
+} /* jerryx_handler_assert_not_equal */
+
+/**
+ * Handler for the assert not strict equal operation.
+ *
+ * @return true - if the two arguments are not strict equal.
+ *         otherwise - this function calls jerryx_fatal.
+ *         Note that in this case the function does not return.
+ */
+static jerry_value_t
+jerryx_handler_assert_not_strict_equal (const jerry_value_t func_obj_val, /**< function object */
+                                        const jerry_value_t this_p, /**< this arg */
+                                        const jerry_value_t args_p[], /**< function arguments */
+                                        const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  return jerryx_handler_assert_equal_helper (func_obj_val, this_p, args_p, args_cnt, JERRYX_OP_NOT_STRICT_EQUAL);
+} /* jerryx_handler_assert_not_strict_equal */
+
+/**
+ * Handler to cause an assertion failure.
+ */
+static jerry_value_t
+jerryx_handler_assert_fail (const jerry_value_t func_obj_val, /**< function object */
+                            const jerry_value_t this_p, /**< this arg */
+                            const jerry_value_t args_p[], /**< function arguments */
+                            const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  (void) func_obj_val; /* unused */
+  (void) this_p; /* unused */
+
+  if (args_cnt > 0)
+  {
+    jerryx_fatal (args_p[0]);
+  }
+
+  jerryx_fatal (jerry_create_null ());
+  return jerry_create_null ();
+} /* jerryx_handler_assert_fail */
+
+/**
+ * Handler function to check if a function throws an error.
+ */
+static jerry_value_t
+jerryx_handler_assert_throws (const jerry_value_t func_obj_val, /**< function object */
+                              const jerry_value_t this_p, /**< this arg */
+                              const jerry_value_t args_p[], /**< function arguments */
+                              const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  (void) func_obj_val; /* unused */
+  (void) this_p; /* unused */
+
+  if (args_cnt > 0 && jerry_value_is_function (args_p[0]))
+  {
+    jerry_value_t value = jerry_call_function (args_p[0], jerry_create_undefined (), NULL, 0);
+
+    if (jerry_value_is_error (value))
+    {
+      jerry_value_t result = jerry_get_value_from_error (value, true);
```

Comment:
This line could be moved right before the binary operation on line 372. The second argument should be `false`, otherwise we might do a double free later.